docs/shared_bindings_matrix.py

```diff
@@ -23,24 +23,47 @@
 
 import json
 import os
+import pathlib
 import re
 import subprocess
 import sys
 
 
 SUPPORTED_PORTS = ['atmel-samd', 'esp32s2', 'litex', 'mimxrt10xx', 'nrf', 'stm']
 
+def get_circuitpython_root_dir():
+    """ The path to the root './circuitpython' directory
+    """
+    cwd = pathlib.Path('.').resolve()
+    cwd_parts = cwd.parts
+
+    root_idx = len(cwd_parts)
+
+    # Search the path from tail to head, so that we capture the
+    # deepest folder. This avoids overshooting in instances like:
+    # '/home/user/circuitpython_v5/circuitpython'
+    for idx, val in enumerate(cwd_parts[::-1]):
+        if val.startswith("circuitpython"):
+            root_idx = root_idx - idx
+            break
```

Comment:
I frequently use directory names that are _NOT_ "circuitpython" for my circuitpython clones.  For instance, a tree which I never put my own changes in I call `cp-pristine`.  Won't this code work improperly in these cases?

Alternative: Use the predefined variable `__file__` in the `shared_bindings_matrix` module, since it is at a fixed location relative to the circuitpython source tree:
```
def get_circuitpython_root_dir():
    p = pathlib.Path(__file__).resolve()
    return p.parent.parent
```

Reply:
Good point on different clone names all together! And I've seen this incantation before...totally forgot about it. Thanks @jepler!